api/src/main/java/org/pmiops/workbench/api/UserMetricsController.java

```diff
@@ -228,27 +229,35 @@ private boolean foundBlobIdsContainsUserRecentlyModifiedResource(
   }
 
   @VisibleForTesting
-  public boolean hasValidBlobIdIfNotebookNamePresent(
-      DbUserRecentlyModifiedResource dbUserRecentResource) {
-    if (dbUserRecentResource.getResourceType()
-        == DbUserRecentlyModifiedResource.DbUserRecentlyModifiedResourceType.NOTEBOOK) {
-      return Optional.ofNullable(dbUserRecentResource.getResourceId())
-          .map(name -> uriToBlobId(name).isPresent())
-          .orElse(true);
+  public boolean isValidResource(DbUserRecentlyModifiedResource resource) {
+    // handle null and unparseable resourceIds
+    final Optional<Long> resourceId =
+        Optional.ofNullable(resource.getResourceId())
+            .flatMap(str -> Optional.ofNullable(Longs.tryParse(str)));
+    switch (resource.getResourceType()) {
+      case COHORT:
+        return resourceId.flatMap(cohortService::findByCohortId).isPresent();
```

Comment:
these are all the no-throw versions of these calls, some of which I'm creating in this PR